wallet/src/main/scala/org/bitcoins/wallet/DLCWallet.scala

```diff
@@ -502,28 +571,41 @@ abstract class DLCWallet extends LockedWallet with UnlockedWalletApi {
         remoteChangeSPK = remoteSetupMsg.changeAddress.scriptPubKey,
         network = dlcOffer.network
       )
+    }
+  }
 
-      val setupF = if (dlcDb.isInitiator) {
-        // TODO: Note that the funding tx in this setup is not signed
-        val cetSigs = CETSignatures(outcomeSigs, dlcAccept.refundSig)
-        client.setupDLCOffer(
-          Future.successful(cetSigs),
-          (_, _) => FutureUtil.unit,
-          Future.successful(client.createUnsignedFundingTransaction))
-      } else {
-        val cetSigs = CETSignatures(outcomeSigs, dlcDb.refundSigOpt.get)
-        val fundingSigs =
-          fundingInputs
-            .filter(_.isInitiator)
-            .map(input => (input.outPoint, input.sigs))
-            .toMap
-        client.setupDLCAccept(
-          _ => FutureUtil.unit,
-          Future.successful((cetSigs, FundingSignatures(fundingSigs))))
-      }
+  private def clientAndSetupFromDb(
+      dlcDb: DLCDb,
+      dlcOffer: DLCOfferDb,
+      dlcAccept: DLCAcceptDb,
+      fundingInputs: Vector[DLCFundingInputDb],
+      outcomeSigDbs: Vector[DLCCETSignatureDb]): Future[(DLCClient, SetupDLC)] = {
 
-      setupF.map((client, _))
-    }
+    clientFromDb(dlcDb, dlcOffer, dlcAccept, fundingInputs, outcomeSigDbs)
+      .flatMap { client =>
+        val outcomeSigs = outcomeSigDbs.map(_.toTuple).toMap
+
+        val setupF = if (dlcDb.isInitiator) {
+          // TODO: Note that the funding tx in this setup is not signed
```

Comment:
not sure what you mean here

Reply:
This comment was there previously, is a note that we could should maybe tell the user that if you're the initiator you don't have the other party's funding sigs.